docs/high_availability_installation_guide.md

```diff
@@ -0,0 +1,284 @@
+
+**Redis**: The purpose of having Redis is to store UI session data and store the registry metadata cache. When one Harbor instance fails or the load balancer routes a user request to another Harbor instance, any Harbor instance can query the Redis to retrieve session information to make sure the end-user has a continued session. User should follow the best practice of Redis to make it HA protect.  
+
+From the above high availability architecture, we can see that we don't setup LB per stateless services. Instead we group those stateless service as a group. The communicate between each services are protected by host based docker network with isolation. **Please Note!** As the component communicate with each other through rest API. You can always define the group granularity according to your use scenarios. 
+
+### Limitation 
+
+Currently Harbor doesn’t support and Notary in HA scenario. Which means it don't support Content trust for this HA setup. 
+
+## Setup 
+
+Follow the setup instruction in this section we build a Harbor high availability deployment as the follow figure shows. You can setup more Harbor nodes if needed. 
+
+![setup](img/ha/LabInstallation.png)
+
+### Prerequisites 
+
+ - 1> MariaDB cluster(Harbor-DB,192.168.1.215, currently harbor use mariadb 10.2.10) 
+ - 2> Shared Storage (Swift Server. 192.168.1.216. **Important!** The shared storage is replaceable you can choose other shared storage, just need to make sure the storage you used is supported by registry https://docs.docker.com/registry/storage-drivers) 
+ - 3> Redis cluster(192.168.1.217)
+ - 4> PostgreSQL (Clair DB 192.168.1.50. Note! The PostgreSQL is optional, it only needed when you use the vulnerability scanning function, currently harbor use PostgreSQL 9.6.5) 
+
+**Important!** Item 1,2,3,4 are considered external components to Harbor. Before configuring Harbor HA, we assume these components are present and all of them are HA protected. Otherwise, any of these components can be a single point of failure. 
+
+**Tips!**
+
+If you are only setting up the HA for POC purpose. You can use docker to run MariaDB, Redis and PostgreSQL in one OS with follow command. 
+```
+#> docker run --name redis-server -p 6379:6379 -d redis 
+#> docker run -d --restart=always -e MYSQL_ROOT_PASSWORD=root123 -v /data/database:/var/lib/mysql:z -p 3306:3306 --name mysql vmware/mariadb-photon:10.2.10 
+#> docker run -d -e POSTGRES_PASSWORD="password" -p 5432:5432 postgres:9.6 
+```
+ - 5> 2 VMs for Load balancer cluster. 
+ - 6> n VMs for Harbor stateless services (n >=2), in this example we will set up 2 Harbor nodes.     
+ - 7> n+1 static IPs (1 for VIP and the other n IPs will be used by Harbor stateless servers)  
+
+### Load balancer setup 
+
+Here we will use Ubuntu16.04 LTS as the Load balancer cluster server. You can choose other Linux OS if needed. 
+
+
+#### On Loadbalancer01: 
+
+ - 1> install the Keepalived application.
+
+```
+    $> apt-get install keepalived 
```

Comment:
"$>" is not needed, and it should be "sudo apt-get install keepalived" or run as root